hq/app/aws/ec2/EC2.scala

```diff
@@ -123,6 +125,36 @@ object EC2 {
       )
   }
 
+  def updateSgsWithTags(account: AwsAccount, sgIds: List[SGOpenPortsDetail])
+                       (sgsDetailsFn: (AmazonEC2Async, DescribeSecurityGroupsRequest) => Attempt[DescribeSecurityGroupsResult])
```

Comment:
Also seems like it's doing quite a few different things. I'ma stare at it and see if it can be simplified.

Reply:
Added the function for the testing purpose, can be changed, however we might lose the test case.
I feel like it'd better to have AWS layer separated since more features are added.